x-pack/plugins/maps/public/connected_components/add_layer_panel/flyout_body/layer_wizard_select.tsx

```diff
@@ -5,26 +5,62 @@
  */
 
 import _ from 'lodash';
-import React, { Component, Fragment } from 'react';
-import { EuiSpacer, EuiCard, EuiIcon } from '@elastic/eui';
-import { EuiLoadingContent } from '@elastic/eui';
+import React, { Component } from 'react';
+import {
+  EuiCard,
+  EuiIcon,
+  EuiFlexGrid,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingContent,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
 import { getLayerWizards, LayerWizard } from '../../../classes/layers/layer_wizard_registry';
+import { LAYER_WIZARD_CATEGORY } from '../../../../common/constants';
 
 interface Props {
   onSelect: (layerWizard: LayerWizard) => void;
 }
 
 interface State {
-  layerWizards: LayerWizard[];
+  activeCategories: LAYER_WIZARD_CATEGORY[];
   hasLoadedWizards: boolean;
+  layerWizards: LayerWizard[];
+  selectedCategory?: LAYER_WIZARD_CATEGORY;
```

Comment:
I'd avoid using undefined to manage known state and having to resort to optional params. It's _not_ optional, because we're setting it explicitly and it drives the UX. We know exactly what values we can expect.  Using `null` instead to avoid any ambiguity. Then you also don;t have weird looking calls like `this._filterByCategory(undefined)` but can use `this._filterByCategory(null)` instead.

e.g.
```
selectedCategory: LAYER_WIZARD_CATEGORY | null
```